shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceImpl.java

```diff
@@ -64,19 +70,25 @@ void syncUpdatesFromManager() {
     public Uni<HttpResponse<Buffer>> notifyBridgeStatusChange(BridgeDTO bridgeDTO) {
         LOGGER.debug("Notifying manager about the new status of the Bridge '{}'", bridgeDTO.getId());
         return getAuthenticatedRequest(webClientManager.put(APIConstants.SHARD_API_BASE_PATH), request -> request.sendJson(bridgeDTO))
+                .onItem().invoke(success -> metricsService.updateManagerRequestMetrics(ManagerRequestType.UPDATE, ManagerRequestStatus.SUCCESS))
+                .onFailure().invoke(failure -> metricsService.updateManagerRequestMetrics(ManagerRequestType.UPDATE, ManagerRequestStatus.FAILURE))
                 .onFailure().retry().withBackOff(WebClientUtils.DEFAULT_BACKOFF).withJitter(WebClientUtils.DEFAULT_JITTER).atMost(WebClientUtils.MAX_RETRIES);
```

Comment:
@vaibhavjainwiz I find this quite hard to follow. The second `onFailure` call, what is going to be retried? The original  call the Fleet Manager, or the attempt to update the `metricsService`?

Reply:
The original call of Fleet Manager will be retry again on failure. Metrics will be published with every failed attempt for call to fleet manager.